.evergreen/config.yml

```diff
@@ -411,71 +427,69 @@ axes:
   - id: "spark"
     display_name: "Spark"
     values:
-      - id: "3.1.2"
-        display_name: "Spark 3.1.2"
+      - id: "3.1.3"
+        display_name: "Spark 3.1.3"
+        variables:
+          SPARK: "3.1.3"
+      - id: "3.2.4"
+        display_name: "Spark 3.2.4"
+        variables:
+          SPARK: "3.2.4"
+      - id: "3.3.4"
+        display_name: "Spark 3.3.4"
         variables:
-          SPARK: "3.1.2"
-      - id: "3.2.2"
-        display_name: "Spark 3.2.2"
+          SPARK: "3.3.4"
+      - id: "3.4.2"
+        display_name: "Spark 3.4.2"
         variables:
-          SPARK: "3.2.2"
+          SPARK: "3.4.2"
+      - id: "3.5.1"
+        display_name: "Spark 3.5.1"
+        variables:
+          SPARK: "3.5.1"
 
 buildvariants:
 
   - matrix_name: "static-checks"
     matrix_spec: { scala: "*", spark: "*" }
     display_name: "Static checks: ${scala} ${spark}"
-    exclude_spec:
-      - {scala: "2.12", spark: "3.2.2"}
-      - {scala: "2.13", spark: "3.1.2"}
-    run_on:
-      - "ubuntu1804-test"
+    <<: *exclude-spec
     tags: ["static-check"]
     tasks:
       - name: "static-checks-task"
 
   - matrix_name: "unit-tests"
     matrix_spec: { scala: "*", spark: "*" }
-    exclude_spec: {scala: "2.13", spark: "3.1.2"}
+    <<: *exclude-spec
     display_name: "Units tests: ${scala} ${spark}"
     tags: ["unit-test"]
-    run_on:
-      - "ubuntu1804-test"
     tasks:
       - name: "unit-test-task"
 
   - matrix_name: "integration-tests-2-12"
-    matrix_spec: { scala: "2.12", spark: "*", version: ["6.0"], topology: "sharded-cluster"}
+    matrix_spec: { scala: "2.12", spark: "3.1.3", version: ["7.0"], topology: "sharded-cluster"}
     display_name: "Integration tests: ${scala} ${spark} ${version} ${topology}"
     tags: ["integration-test"]
-    run_on:
-      - "ubuntu1804-test"
     tasks:
       - name: "integration-test-task"
 
   - matrix_name: "integration-tests-2-13"
-    matrix_spec: { scala: "2.13", spark: "3.2.2", version: "*", topology: "*"}
+    matrix_spec: { scala: "2.13", spark: "*", version: "*", topology: "*"}
```

Comment:
What's the rationale for the variants for which we run integration tests?  For instance, I'm unclear why we run integration tests with Spark 3.1, 3.2, and 3.5, but not 3.3 and 3.4.   Or why we only run tests against latest sharded clusters but all MongoDB releases for replica sets.

Reply:
I've updated this as it wasn't clear. The rationale should be to reduce the matrix to cover the requirements.

Scala 2.12 is tested against the only Spark version that supports it.
Scala 2.13 is run against the minmum and highest supported versions of Spark and the minimum and highest versions of MongoDB.  

The latest version of MongoDB is also included to act as a canary.